addons/product/models/product_product.py

```diff
@@ -237,7 +237,7 @@ def _check_barcode_uniqueness(self):
         to ensure the uniqueness between products' barcodes and packagings' ones"""
         # Barcodes should only be unique within a company
         for company_id, barcodes_within_company in self._get_barcodes_by_company():
-            self._check_duplicated_product_barcodes(barcodes_within_company, company_id)
+            self.with_context(skip_preprocess_gs1=True)._check_duplicated_product_barcodes(barcodes_within_company, company_id)
```

Comment:
It seems odd to me in a method that explicitly says *why* it should check barcodes considering the GS1 nomenclature to disable the pre-process of the GS1 barcodes :thinking: